pkg/gatherers/clusterconfig/service_accounts.go

```diff
@@ -96,32 +96,35 @@ func gatherServiceAccounts(ctx context.Context, coreClient corev1client.CoreV1In
 
 // ServiceAccountsMarshaller implements serialization of Service Accounts
 type ServiceAccountsMarshaller struct {
-	sa                   []corev1.ServiceAccount
+	serviceAccounts      []corev1.ServiceAccount
 	totalServiceAccounts int
 }
 
 // Marshal implements serialization of ServiceAccount
 func (a ServiceAccountsMarshaller) Marshal() ([]byte, error) {
 	// Creates map for marshal
-	sr := map[string]interface{}{}
-	st := map[string]interface{}{}
-	st["TOTAL_COUNT"] = a.totalServiceAccounts
-	sr["serviceAccounts"] = st
-	nss := map[string]interface{}{}
-	st["namespaces"] = nss
-	for i := range a.sa {
-		var ns map[string]interface{}
-		var ok bool
-		if _, ok = nss[a.sa[i].Namespace]; !ok {
-			ns = map[string]interface{}{}
-			nss[a.sa[i].Namespace] = ns
-		} else {
-			ns = nss[a.sa[i].Namespace].(map[string]interface{})
+	serviceAccounts := map[string]any{}
+	namespaces := map[string][]map[string]any{}
+	serviceAccounts["serviceAccounts"] = map[string]any{
+		"TOTAL_COUNT": a.totalServiceAccounts,
+		"namespaces":  namespaces,
+	}
+
+	for i := range a.serviceAccounts {
+		serviceAccount := []map[string]any{
+			{
+				"name":    a.serviceAccounts[i].Name,
+				"secrets": len(a.serviceAccounts[i].Secrets),
+			},
+		}
+		_, ok := namespaces[a.serviceAccounts[i].Namespace]
+		if !ok {
+			namespaces[a.serviceAccounts[i].Namespace] = serviceAccount
+			continue
```

Comment:
I think we don't need this. Do we?

Reply:
I think that it does make sense. At line 125 it will append a new SA to the namespace map with the SA index. So, I'm assuming that we have > 1 SA per namespace, which means that you need first to create the index in the namespace map before appending it.

But probably what we don't need is line 114 to be a map as well...

Reply:
```go
// here comes the custom type that you suggested
serviceAccount := map[string]any{
    "name":    a.serviceAccounts[i].Name,
    "secrets": len(a.serviceAccounts[i].Secrets),
}
_, ok := namespaces[a.serviceAccounts[i].Namespace]
if !ok {
    namespaces[a.serviceAccounts[i].Namespace] = serviceAccount
    continue
}
namespaces[a.serviceAccounts[i].Namespace] = append(namespaces[a.serviceAccounts[i].Namespace], serviceAccount)
```

Reply:
@rhrmo my fault here, that one should not be a slice. It is hard to write code directly to GitHub. lol

Reply:
We need it to be slice of maps, because one namespace can have multiple service accounts, but I like the idea of introducing the type for the service account so it will get more clear after I update it.

Reply:
nvm I was wrong, it does actually work without it being a slice and without ok